src/Marketplace/Controller.php

```diff
@@ -502,6 +502,18 @@ public static function hasWriteAccess(): bool
         return is_dir(GLPI_MARKETPLACE_DIR) && is_writable(GLPI_MARKETPLACE_DIR);
     }
 
+    /**
+     * Check if a directory ".git" is present in plugins source code.
+     *
+     * @param string $plugin_key
+     *
+     * @return bool
+     */
+    public static function hasGitDirectory(string $plugin_key = ""): bool
+    {
+        return is_dir(GLPI_MARKETPLACE_DIR . "/" . $plugin_key . "/.git");
+    }
```

Comment:
1. `$plugin_key` param is not optionnal, so it should not have a default value.

2. Not everyone uses Git. Maybe we sould check for some other popular VCS.

```suggestion
    public static function hasVcsDirectory(string $plugin_key): bool
    {
        $vcs_dirs = [
            '.git',
            '.hg', // Mercurial
            '.svn',
        ];
        foreach ($vcs_dirs as $vcs_dir) {
            if (is_dir(GLPI_MARKETPLACE_DIR . '/' . $plugin_key . '/' . $vcs_dir)) {
                return true;
            }
        }

        return false;
    }
```